style_editor.html

```diff
@@ -9391,87 +9403,101 @@
+        const actions = {};
+
+        /* Add values from the enum builder to an array and sort alphabetically,
+        excluding effects with dedicated buttons.*/
+        const values = Object.entries(EFFECT_ENUM_BUILDER.value_to_name)
+          .sort((a, b) => a[1].localeCompare(b[1]))
+          .filter(([value]) => ![
+            EFFECT_CLASH,
+            EFFECT_STAB,
+            EFFECT_BLAST,
+            EFFECT_FORCE,
+            EFFECT_BOOT,
+            EFFECT_NEWFONT,
+            EFFECT_PREON,
+          ].includes(Number(value)));
+
+        // Add sub-groups for the different categories of effects
+        const generalEffectsMenu = document.createElement('optgroup');
+        generalEffectsMenu.label = 'General Effects';
+        const userEffectsMenu = document.createElement('optgroup');
+        userEffectsMenu.label = 'User Effects';
+        const blasterEffectsMenu = document.createElement('optgroup');
+        blasterEffectsMenu.label = 'Blaster Effects';
+        const gameEffectsMenu = document.createElement('optgroup');
+        gameEffectsMenu.label = 'Game Effects';
+        const errorMessagesMenu = document.createElement('optgroup');
+        errorMessagesMenu.label = 'Error Messages';
+
+        // Add sorted values to the menu and actions dictionary,
+        for (const [value, name] of values) {
+          const nameWithoutEffect = name.replace(/^EFFECT_/, '');
+          const option = document.createElement('option');
+          option.value = value;
+          option.text = nameWithoutEffect;
+
+          // Check if the effect belongs to a certain category and add it to the corresponding sub-group
+          if (name.startsWith('EFFECT_GAME')) {
+            gameEffectsMenu.appendChild(option);
+          } else if (name.startsWith('EFFECT_USER')) {
+            userEffectsMenu.appendChild(option);
+          } else {
+            switch (Number(value)) {
+              case EFFECT_STUN:
+              case EFFECT_FIRE:
+              case EFFECT_CLIP_IN:
+              case EFFECT_CLIP_OUT:
+              case EFFECT_RELOAD:
+              case EFFECT_MODE:
+              case EFFECT_RANGE:
+              case EFFECT_EMPTY:
+              case EFFECT_FULL:
+              case EFFECT_JAM:
+              case EFFECT_UNJAM:
+              case EFFECT_PLI_ON:
+              case EFFECT_PLI_OFF:
+                blasterEffectsMenu.appendChild(option);
+                break;
+              case EFFECT_ERROR_IN_BLADE_ARRAY:
+              case EFFECT_ERROR_IN_FONT_DIRECTORY:
+              case EFFECT_FONT_DIRECTORY_NOT_FOUND:
+              case EFFECT_SD_CARD_NOT_FOUND:
+              case EFFECT_LOW_BATTERY:
+                errorMessagesMenu.appendChild(option);
+                break;
+              default:
+                generalEffectsMenu.appendChild(option);
+                break;
+            }
+          }
+          actions[value] = function() { blade.addEffect(value, 0.0) };
```

Comment:
didn't we remove this in the other PR?

Reply:
Indeed. Fixed.
This is where multiple PRs going out of sequence are my hurdle. 
I might need to do one at a time to make sure I'm comparing to the current.

Reply:
It can be a problem sometimes.
Ideally parallel PRs should touch separate parts of the code, then it's easy.

Reply:
Right now github says there are conflicts that needs to be resolved before this change can be merged.